shell.nix

```diff
@@ -1,11 +1,12 @@
 let
-  inherit (import ./default.nix {}) haskellPkgs;
+  inherit (import ./default.nix {}) haskellPkgs pkgs;
 in
 
 haskellPkgs.shellFor {
   packages = p: [ p.repro ];
-  nativeBuildInputs = [ haskellPkgs.cabal-install ];
+  nativeBuildInputs = [ pkgs.cabal-install ];
   shellHook = ''
     ghc-pkg list
   '';
+  genericBuilderArgsModifier = args: args // { doCheck = true; };
```

Comment:
This is functionality added in https://github.com/cdepillabout/nixpkgs/commit/47d19bef86481df4f52c2bf909c23940cf98973a, but it is not yet in nixpkgs.

We'll probably discuss what to do about this in https://github.com/NixOS/nixpkgs/issues/104349.